File: Services/Venue/Venue.TimeCafe.Domain/Models/Tariff.cs
namespace Venue.TimeCafe.Domain.Models;

public class Tariff
{
    public Tariff(Guid themeId)
    {
        TariffId = themeId;
    }

    public Tariff()
    {
        TariffId = Guid.NewGuid();
    }

    public Guid TariffId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public decimal PricePerMinute { get; set; }
    public BillingType BillingType { get; set; }
    public Guid? ThemeId { get; set; }

    public string? Summary { get; set; }
    public List<string> Features { get; set; } = [];
    public List<string> AudienceTags { get; set; } = [];
    public int? MinSessionMinutes { get; set; }
    public string? RoundingRule { get; set; }
    public int? MaxGuests { get; set; }
    public string? CancellationPolicy { get; set; }
    public bool IsRecommended { get; set; }
    public int SortOrder { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset LastModified { get; set; } = DateTimeOffset.UtcNow;

    public virtual ICollection<Visit> Visits { get; set; } = [];
}

File: WebApp/timecafe.react.ui/src/types/tariff.ts
export const BillingType = {
    Hourly: 1,
    PerMinute: 2,
} as const;

export type BillingType = (typeof BillingType)[keyof typeof BillingType];

export type CostBreakdown = {
    actualMinutes: number;
    billableMinutes: number;
    baseCost: number;
    finalCost: number;
    optimizationGain: number;
};

export type Tariff = {
    tariffId: string;
    name: string;
    description: string;
    billingType: BillingType;
    pricePerMinute: number;
    isActive: boolean;
    themeId?: string | null;

    accent?: "brand" | "green" | "pink" | "purple";
    recommended?: boolean;

    themeName?: string;
    themeEmoji?: string | null;
    themeColors?: string | null;
    colors?: string | null;

    summary?: string | null;
    features?: string[] | null;
    audienceTags?: string[] | null;
    minSessionMinutes?: number | null;
    roundingRule?: string | null;
    maxGuests?: number | null;
    cancellationPolicy?: string | null;
    isRecommended?: boolean;
    sortOrder?: number;
    calculationExamples?: CostBreakdown[] | null;
};
